### ISSUE TITLE
Facet update. Wrong payload

### ISSUE THREAD

**Issue by ivan-kleshnin** [Yomguithereal/baobab#296]
Related to https://github.com/Yomguithereal/baobab/issues/293

```js
someCursor.on("update", function (x) {
  console.log(x.data); // +
});

someFacet.on("update", function (x) {
  console.log(x.data); // -
});
```

**Comment by Yomguithereal** [comment]
Pending #293 decisions.

**Comment by Yomguithereal** [comment]
I guess this is v1?

**Comment by ivan-kleshnin** [comment]
Yes. Feel free to close if it's fixed in V2.

**Comment by Yomguithereal** [comment]
Since facets and cursors share the same API in v2 this should be fixed indeed. But let me check a final time just to be sure.

**Comment by ivan-kleshnin** [comment]
Payload is present but contains wrong `previousData` for facets.

##### Proof sandbox

Facet:
```js
$firstItem: [
  ["items", 0],
  function (item) {
    return item;
  }
]
```

Source code:
https://gist.github.com/ivan-kleshnin/ed6d10ba945b3bf4ca01

Screenshot:
![prevdata bug](https://cloud.githubusercontent.com/assets/2128182/8773145/4a372648-2ed2-11e5-8f9d-0b23ca4c0ce9.png)

What `data.previousData` contains depends on the facet declaration syntax. Here it's provided with an Array (LISP) style. With Object style it was different but wrong nevertheless.

If you change facet declaration to

Facet:
```js
$firstItem: [
  ["items"],
  function (items) {
    return items[0];
  }
]
```

`data.previousData` is still the same.

**Comment by Yomguithereal** [comment]
Ok, I see where it went wrong. I need to fix this.

**Comment by Yomguithereal** [comment]
I will probably need to make the `currentData` and `previousData` keys js getters so that the remain lazy. Else they would be computed at each update of the tree and this is not strictly necessary and it would render facets laziness inoperant.

**Comment by ivan-kleshnin** [comment]
Sounds reasonable.

**Comment by Yomguithereal** [comment]
This should be fixed as per `2.0.0-dev12`.

**Comment by ivan-kleshnin** [comment]
Fixed.
